Extensions/timestamps.js

```diff
@@ -224,12 +245,91 @@ XKit.extensions.timestamps = new Object({
 
 			var timestamp = responseData.posts[0].timestamp;
 			date_element.html(this.format_date(moment(new Date(timestamp * 1000))));
-			date_element.removeClass("xtimestamp_loading");
+			date_element.removeClass("xtimestamp-loading");
 			XKit.storage.set("timestamps", "xkit_timestamp_cache_" + post_id, timestamp);
 		})
 		.catch(() => this.show_failed(date_element));
 	},
 
+	react_add_timestamps: function() {
+		var posts = $("[data-id]:not(.xkit_timestamps)");
+		posts.addClass("xkit_timestamps");
+
+		if (posts.length === 0) {
+			return;
+		}
+
+		XKit.extensions.timestamps.check_quota();
+
+		posts.each(function() {
+			var post = $(this);
+
+			var post_id = post.attr("data-id");
+
+			var xtimestamp_class = "xtimestamp";
+			if (XKit.extensions.timestamps.in_search) {
+				xtimestamp_class = "xtimestamp-in-search";
+			}
+
+			var xtimestamp_html = `<div class="xkit_timestamp_${post_id} ${xtimestamp_class} xtimestamp-loading">&nbsp;</div>`;
+			$(xtimestamp_html).insertAfter(post.find("header"));
+
+			var note = $(".xkit_timestamp_" + post_id);
+			XKit.extensions.timestamps.react_fetch_timestamp(post_id, note);
+		});
+	},
+
+	react_add_reblog_timestamps: function() {
+		var reblogs = $(XKit.extensions.timestamps.reblogs_class).not(".xkit_timestamps");
+
+		if (XKit.extensions.timestamps.preferences.reblogs.value === "op") {
+			var posts = $("[data-id]");
+			reblogs = posts.map(function() { return $(this).children(XKit.extensions.timestamps.reblogs_class).not(".xkit_timestamps").get(0); });
+		}
+
+		reblogs
+		.addClass("xkit_timestamps")
+		.each(function() {
+			var reblog = $(this);
+
+			if (reblog.length === 0) { return; }
```

Comment:
when would `.each()` pass in a nonexistent element?

Reply:
Ah, ur suggesting an early return like `if (reblogs.length === 0) { return; }`, right?

Reply:
if `reblogs` is an empty set, then both `.addClass` and `.each` just won't do anything, so it's redundant